src/language-html/utils.js

```diff
@@ -287,7 +287,6 @@ function forceBreakContent(node) {
       (["body", "script", "style"].includes(node.name) ||
         node.children.some((child) => hasNonTextChild(child)))) ||
     (node.firstChild &&
-      node.firstChild === node.lastChild &&
```

Comment:
Any idea on why we had this condition?

Reply:
Don't know, but I think this mean to test text node is the only node?

Reply:
These should has no difference.

**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAdKACDAeAJgSwDcMwAbAQwGdKBeVEAJzl3oD50ssB1ODcpjAHMI+KIIwwIfXLgkALfJQyV8AWwAOpXmH6y5cJhxwB6AoXaYcZkhWp1GzNke5zyMDIqEixGaDkrq5FCs6gwQ6gCM2MYBQRZY0WYWIAA0IOEw+NCUyKD8YQDuAAr8CDko5KQF5ACeOWkARgzkYADWcDAAyoFgooLIMAwArnBpcjCqpFwK8LFgcJ1l+JmEyzXI4NSpIKKUBjBFzYKq5MgAZpV7aQBWlAAeAELNbR2d5KpwADKicOeXoyBbndOn0tABFIYQeB-UhXECBBh7BgbcaTbahUQwLj4XAwOTIAAcAAY0qEIHsuM11BtQnAkYRfmkAI6Q+CHcLlEBUAC0UDgzGY2yYLPwTEO5GOpyQF1hAL2qnwA2GctBcAhUN+0v+aRg5Aa2Nx+KQACYdc18KQ+gBhCCqE4bOkAVm2Qz2ABU9eUZXDCCMAJJQXAILpgBj4dQwACCgc6MBqWhhewAvkmgA)
```sh
--parser html
```

**Input:**
```html

  <div class="red">
    We are going to add this simple card here
  </div>
  <div class="red">
    What is going on <span>prop1</span>
  </div>

```

**Output:**
```html
<div class="red">
  We are going to add this simple card here
</div>
<div class="red">What is going on <span>prop1</span></div>

```

**Prettier pr-8614**
[Playground link](https://deploy-preview-8614--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAdKACDAeAJgSwDcMwAbAQwGdKBeVEAJzl3oD50ssB1ODcpjAHMI+KIIwwIfXLgkALfJQyV8AWwAOpXmH6y5cJhxwB6AoXaYcZkhWp1GzNke5zyMDIqEixGaDkrq5FCs6gwQ6gCM2MYBQRZY0WYWIAA0IOEw+NCUyKD8YQDuAAr8CDko5KQF5ACeOWkARgzkYADWcDAAyoFgooLIMAwArnBpcKoNzLjMADJBgkPkgnAAYhAMqm6ZYsgg5EOSqSByMKqkXArwsWBwnWX4mYQPNbtg1EeilAYwRc2Cm8gAGaVL5pABWlAAHgAhZptDqdciqOAzURwIEg0YgCGQzp9LQARSGEHgGNIoJAgQYXwYuxOZyOoVEMC4+FwMDkyAAHAAGNKhCBfLjNdS7UJwGmEdFpACOxPgv3C5T2lAAtFA4MxmEcmHL8ExfksAUhgeSsV9VPgBsNzfi4ESSeiTZi0jByA1WezOUgAEyu5r4Uh9ADCEFUxpAEoArEchl8ACru8qmimEEYASSg01gnTADHw6hgAEEs50YDUtGSvgBfatAA)
```sh
--parser html
```

**Input:**
```html

  <div class="red">
    We are going to add this simple card here
  </div>
  <div class="red">
    What is going on <span>prop1</span>
  </div>

```

**Output:**
```html
<div class="red">
  We are going to add this simple card here
</div>
<div class="red">
  What is going on <span>prop1</span>
</div>

```

Reply:
https://github.com/prettier/prettier/pull/5596/files

Reply:
#5596 fixed issues with line breaks around elements. Do we really need to preserve line breaks around text?

Are we sure the following output wouldn't be better in #8287?
```html
<template>
  <div class="red">We are going to add this simple card here</div>
  <div class="red">What is going on {{ prop1 }} and {{ prop2 }}</div>
</template>
```

Reply:
Seems we are checking leading line breaks

**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAdKACDAeAJgSwDcMwAbAQwGdKBeVEAJzl3oD4B1ODcpjAcwj4ofDDAjdcuUQAt8lDJXwBbAA6kuYHlOlwm6dNgD0BQq3RY8REhWp1GzNu2nkYGOf0HCM0HJRXkoVhUGCBUARiM-ALMoA2MiGMtiMipaeiYWEA5nV3cBIREfYGAMYNCwjABfSu4oKWLSkJUAJirK8ww4kxiQABoQUJh8aEpkUB4QgHcABR4EUZRyUknyAE9R-oAjBnIwAGs4GABlfzAC5BgGAFc4fukYJVInfHgosDgj+ZeiF9XkcGofRAQkouhg0x2fCU5GQADMlqD+gArSgADwAQjt9ocjuQlHAADJCOBwhG3EAo1FHArqACKVwg8FJpERIH8DFBDH+90eQOCQhg7HwuBg0mQAA4AAz9Mqg9g7FT-YJwTmEEn9ACODPgENCCxAVAAtFA4MxmECmFr8EwIeQoTCkPCWeTQUp8Bdri6aXB6YySY6yf0YORNkKRWKkM0gzt8KQCgBhCBKaH-FUAViBV1BABUQwsnazCDcAJJ1BDHMAMfAqGAAQTqRxgq3UzNB1SAA)
```sh
--parser html
```

**Input:**
```html

  <div class="red">We are going to add this simple card here

</div>
  <div class="red">What is going on <span>prop1</span>

</div>
<div class="red">What is going on {{ prop1 }} and {{ prop2 }}
  
</div>

```

**Output:**
```html
<div class="red">We are going to add this simple card here</div>
<div class="red">What is going on <span>prop1</span></div>
<div class="red">What is going on {{ prop1 }} and {{ prop2 }}</div>

```

Reply:
Yes, see https://github.com/prettier/prettier/pull/5596#issue-236042241

Reply:
The question is: do we want to do that for text (with or without interpolations)? It was done to solve issues with elements, not with text.

Reply:
What about this `<div class="red">What is going on <span>prop1</span></div>`. It's element.

Reply:
Could be a icon `<div class="red">What is going on <i class="icon"></i></div>`

Reply:
I mean the cases where the element contains only text (probably with interpolations), like in #8287. If there are elements, we can keep applying the rules from https://github.com/prettier/prettier/pull/5596#issue-236042241. But if there is only text?

Reply:
I'll check.

Reply:
I've updated, but feel less readable for long text.